src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3752,16 +3759,31 @@ private enum PostSkipAction
                     accessorKind = SyntaxKind.UnknownAccessorDeclaration;
                 }
 
-                BlockSyntax body = null;
+                BlockSyntax blockBody = null;
+                ArrowExpressionClauseSyntax expressionBody = null;
+
                 SyntaxToken semicolon = null;
                 bool currentTokenIsSemicolon = this.CurrentToken.Kind == SyntaxKind.SemicolonToken;
-                if (this.CurrentToken.Kind == SyntaxKind.OpenBraceToken || (validAccName && !currentTokenIsSemicolon && !IsTerminator()))
+                bool currentTokenIsArrow = this.CurrentToken.Kind == SyntaxKind.EqualsGreaterThanToken;
+
+                if (this.CurrentToken.Kind == SyntaxKind.OpenBraceToken
+                            || (validAccName && !currentTokenIsSemicolon && !currentTokenIsArrow && !IsTerminator()))
+                {
+                    blockBody = this.ParseBlock(isMethodBody: true, isAccessorBody: true);
+                }
+                else if (this.CurrentToken.Kind == SyntaxKind.EqualsGreaterThanToken)
                 {
-                    body = this.ParseBlock(isMethodBody: true, isAccessorBody: true);
+                    expressionBody = this.ParseArrowExpressionClause();
+                    expressionBody = CheckFeatureAvailability(expressionBody, MessageID.IDS_FeatureExpressionBodiedMethod);
```

Comment:
Added `IDS_FeatureExpressionBodiedDeOrConstructor` and `IDS_FeatureExpressionBodiedAccessor` with corresponding checks in `ParseDestructorDeclaration(...)` and `ParseConstructorDeclaration(...)` after `this.ParseBlockAndExpressionBodiesWithSemicolon(out body, out expressionBody, out semicolon);`. Updated `ParseAccessorDeclaration` respectively.